### PR TITLE
Speed up multigrid NDT align

### PR DESCRIPTION
**Pull request by anhnv3991**
- Speed up multigrid NDT align by removing duplicated processing and replace deep copy by reference
- Should be merge _before_  https://github.com/tier4/ndt_omp/pull/42

### CODE REVIEW

File: include/multigrid_pclomp/multigrid_ndt_omp_impl.hpp [hunk 1]
```diff
@@ -987,46 +1004,14 @@ double pclomp::MultiGridNormalDistributionsTransform<PointSource, PointTarget>::
...
-  for(size_t idx = 0; idx < trans_cloud.size(); ++idx) {
-    PointSource x_trans_pt = trans_cloud[idx];
-
-    // Find neighbors (Radius search has been experimentally faster than direct neighbor checking.
-    std::vector<TargetGridLeafConstPtr> neighborhood;
-    std::vector<float> distances;
 
-    // Neighborhood search method other than kdtree is disabled in multigrid_ndt_omp
-    target_cells_.radiusSearch(x_trans_pt, resolution_, neighborhood, distances);
+    double tmp_score = 0;
```
Comment (SakodaShintaro) [inline include/multigrid_pclomp/multigrid_ndt_omp_impl.hpp:999]: Is this variable used correctly?

**Review by SakodaShintaro — COMMENTED** [review]
It seems to be steadily speeding up. Thank you very much.

elapsed_milliseconds

Since the speed has improved, I also updated result.csv.

File: include/multigrid_pclomp/multigrid_ndt_omp_impl.hpp (hunk 1 above)
Comment (anhnv3991) [reply include/multigrid_pclomp/multigrid_ndt_omp_impl.hpp:999]: @SakodaShintaro I fixed it.

**Review by SakodaShintaro — APPROVED** [review]
Looks Good To Me!

**Review by SakodaShintaro — CHANGES_REQUESTED** [review]
Sorry, I'd like to check it out a little more, so please let me withdraw my approval.

**Review by SakodaShintaro — APPROVED** [review]
I carefully checked the operation. It's OK!

**Pull request merged**